src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -372,9 +372,17 @@ static ucs_status_t
 uct_cuda_ipc_mem_reg(uct_md_h md, void *address, size_t length,
                      const uct_md_mem_reg_params_t *params, uct_mem_h *memh_p)
 {
+    uct_mem_h base = (params != NULL) ?
+                        UCT_MD_MEM_REG_FIELD_VALUE(params, memh, FIELD_MEMH, NULL) :
```

Comment:
`UCT_MD_MEM_REG_FIELD_VALUE(params, memh, FIELD_MEMH, NULL)` is needed only if `ENABLE_PARAMS_CHECK`.
So, you can move in under `if (ENABLE_PARAMS_CHECK)` block.
Moreover,  you can use the following macro:
```C
UCT_CHECK_PARAM(UCT_MD_MEM_REG_FIELD_VALUE(params, memh, FIELD_MEMH, NULL) == NULL,
                "CUDA IPC does not support derived memory handles");
```